clients/admin-ui/src/features/system/AddMultipleSystems.tsx

```diff
@@ -0,0 +1,252 @@
+  const { isOpen, onClose, onOpen } = useDisclosure();
+
+  const allRowsAdded = dictionaryOptions.every((d) => d.linked_system);
+  const columns = useMemo(
+    () => [
+      columnHelper.display({
+        id: "select",
+        header: ({ table }) => (
+          <IndeterminateCheckboxCell
+            {...{
+              checked: table.getIsAllRowsSelected(),
+              indeterminate:
+                table
+                  .getSelectedRowModel()
+                  .rows.filter((r) => !r.original.linked_system).length > 0,
+              onChange: table.getToggleAllRowsSelectedHandler(),
+              manualDisable: allRowsAdded,
+            }}
+          />
+        ),
+        cell: ({ row }) => (
+          <IndeterminateCheckboxCell
+            {...{
+              checked: row.getIsSelected(),
+              disabled: !row.getCanSelect(),
+              indeterminate: row.getIsSomeSelected(),
+              onChange: row.getToggleSelectedHandler(),
+              initialValue: row.original.linked_system,
+            }}
+          />
+        ),
+        meta: {
+          width: "55px",
+        },
+      }),
+      columnHelper.accessor((row) => row.legal_name, {
+        id: "legal_name",
+        cell: (props) => <DefaultCell value={props.getValue()} />,
+        header: (props) => <DefaultHeaderCell value={systemText} {...props} />,
+      }),
+      columnHelper.accessor((row) => row.vendor_id, {
+        id: "vendor_id",
+        cell: (props) => <VendorSourceCell value={props.getValue()} />,
+        header: (props) => <DefaultHeaderCell value="Source" {...props} />,
+        meta: {
+          width: "80px",
+        },
+      }),
+    ],
+    [allRowsAdded, systemText]
+  );
+
+  const rowSelection = useMemo(() => {
+    const innerRowSelection: Record<string, boolean> = {};
+    dictionaryOptions.forEach((ds, index) => {
+      if (ds.linked_system) {
+        innerRowSelection[index] = true;
+      }
+    });
+    return innerRowSelection;
+  }, [dictionaryOptions]);
+
+  const tableInstance = useReactTable<MultipleSystemTable>({
+    columns,
+    data: dictionaryOptions,
+    getCoreRowModel: getCoreRowModel(),
+    getSortedRowModel: getSortedRowModel(),
+    getFilteredRowModel: getFilteredRowModel(),
+    getPaginationRowModel: getPaginationRowModel(),
+    onGlobalFilterChange: setGlobalFilter,
+    enableRowSelection: true,
+    enableSorting: true,
+    enableGlobalFilter: true,
+    state: {
+      globalFilter,
+    },
+    initialState: {
+      rowSelection,
+      pagination: {
+        pageSize: 25,
+      },
+    },
+  });
+
+  const addVendors = async () => {
+    const vendorIds = tableInstance
+      .getSelectedRowModel()
+      .rows.filter((r) => !r.original.linked_system)
+      .map((r) => r.original.vendor_id);
+    if (vendorIds.length > 0) {
+      await postVendorIds(vendorIds);
+      router.push(redirectRoute);
+      toast(
+        successToastParams(
```

Comment:
should we handle the error case too?

Reply:
we should just in case. I'll update it :+1: